src/server.py

```diff
@@ -362,6 +383,16 @@ def __init__(self, metadata):
         self._watcher_by_bus_name = {}
         # Only useful for sounds tagged with "overlap-behavior":
         self._uuid_by_event_id = {}
+        self._background_players = []
+
+    def play(self, uuid_):
+        player = self.players[uuid_]
+        if player.type_ == "bg":
+            # Pause the most recent player.
+            if len(self._background_players) > 0:
+                self._background_players[-1].pause()
+            self._background_players.append(player)
+        player.play()
```

Comment:
Check with David about this, but I think the players need to crossfade when switching from one background music to another, _unless the new background music is just starting_. The duration of this crossfade should probably be independent of the individual sounds' fade in and out times.

I think this because I'm pretty sure we don't want the fade times to be unequal, and we also don't want the resuming background music to start abruptly in the middle of the loop if its defined fade time is 0.

But check with David how he wants to see this exactly.